src/Microsoft.ML.KMeansClustering/KMeansPlusPlusTrainer.cs

```diff
@@ -95,58 +115,31 @@ public class Arguments : UnsupervisedLearnerInputBaseWithWeight
         /// Initializes a new instance of <see cref="KMeansPlusPlusTrainer"/>
         /// </summary>
         /// <param name="env">The <see cref="IHostEnvironment"/> to use.</param>
-        /// <param name="featureColumn">The name of the feature column.</param>
-        /// <param name="weights">The name for the optional column containing the example weights.</param>
-        /// <param name="advancedSettings">A delegate to apply all the advanced arguments to the algorithm.</param>
-        /// <param name="clustersCount">The number of clusters.</param>
-        public KMeansPlusPlusTrainer(IHostEnvironment env,
-            string featureColumn = DefaultColumnNames.Features,
-            int clustersCount = Defaults.K,
-            string weights = null,
-            Action<Arguments> advancedSettings = null)
-            : this(env, new Arguments
-            {
-                FeatureColumn = featureColumn,
-                WeightColumn = weights,
-                K = clustersCount
-            }, advancedSettings)
-        {
-        }
-
-        internal KMeansPlusPlusTrainer(IHostEnvironment env, Arguments args)
-            : this(env, args, null)
+        /// <param name="options">The advanced arguments of the algorithm.</param>
```

Comment:
> /// <param name="options">The advanced arguments of the algorithm.</param> [](start = 7, length = 75)

from KMeansCatalog.cs
` /// <param name="options">Algorithm advanced settings.</param>`
I like mix of arguments, options and settings. Anyone who is favor of one of them get something :)

Reply:
inclusive !

---
In reply to: [248406901](https://github.com/dotnet/machinelearning/pull/2158#discussion_r248406901) [](ancestors = 248406901)